Tools/ros2/ardupilot_sitl/CMakeLists.txt

```diff
@@ -0,0 +1,120 @@
+cmake_minimum_required(VERSION 3.20)
+project(ardupilot_sitl)
+
+# --------------------------------------------------------------------------- #
+# find dependencies
+find_package(ament_cmake REQUIRED)
+find_package(ament_cmake_python REQUIRED)
+
+# --------------------------------------------------------------------------- #
+# add custom targets to configure and build ardupilot
+
+# TODO(srmainwaring): add cache variables for vehicle type, debug etc. 
+# set(ARDUPILOT_VEHICLE "copter" CACHE STRING "Vehicle type")
+
+# NOTE: look for `waf` and set source and build directories to top level.
+# ${PROJECT_SOURCE_DIR} = ./Tools/ros2/ardupilot
+# 
+cmake_path(SET ARDUPILOT_ROOT NORMALIZE ${PROJECT_SOURCE_DIR}/../../..)
+# message("ARDUPILOT_ROOT: ${ARDUPILOT_ROOT}")
+
+find_program(WAF_COMMAND waf HINTS ${ARDUPILOT_ROOT})
+message(STATUS "WAF_COMMAND: ${WAF_COMMAND}")
+
+add_custom_target(ardupilot_configure ALL
+  ${WAF_COMMAND} configure --board sitl --enable-dds --debug
```

Comment:
Do we want to be testing in debug or release? I'd worry if we test in debug, stuff that might be optimized out or behave differently is not caught by CI. Perhaps set `--debug` only if CMAKE_BUILD_TYPE=DEBUG?

Reply:
Yes - release. There are a few macOS specific items I need to swap out. The SITL release build on macOS has been know to seg fault in the past which is why these are here. It should be release every where though. Will change.